sentry_streams/sentry_streams/dummy/dummy_adapter.py

```diff
@@ -10,26 +24,55 @@
 class DummyAdapter(StreamAdapter[DummyInput, DummyOutput]):
     """
     An infinitely scalable adapter that throws away all the data it gets.
+    The adapter tracks the 'streams' that each step returns in the form of
+    lists of previous step names.
     """
 
     def __init__(self, _: PipelineConfig) -> None:
-        pass
+        self.input_streams: MutableMapping[str, List[str]] = {}
+
+    def add_step_input_streams(self, step: Any) -> None:
```

Comment:
Please give this method a more self explanatory name.
What does it do ?
`add_step_input_streams` 

Also step is probably not Any.